aws-xray-client-persistent/library/Network/AWS/XRayClient/Persistent.hs

```diff
@@ -47,7 +49,7 @@ xraySqlBackend sendTrace stdGenIORef subsegmentName =
       , connBegin = binaryTimerWrapper "BEGIN" (connBegin backend)
       , connCommit = unaryTimerWrapper "COMMIT" (connCommit backend)
       , connRollback = unaryTimerWrapper "ROLLBACK" (connRollback backend)
-      , connStmtMap = newConnStmtMap
+      , connStmtMap = mkStatementCache (mkSimpleStatementCache newConnStmtMap)
```

Comment:
You could do it in a compatible way with some pretty minor CPP. I don't remember the _exact_ syntax, but it'd be something like,
```hs
    , connStmtMap = mkCache newConnStmtMap

-- ...


#if MIN_persistent(2, 13, 3)
mkCache :: IORef (Map Text Statement) -> StatementCache
mkCache = mkStatementCache . mkSimpleStatementCache
#else
mkCache :: IORef (Map Text Statement) -> IORef (Map Text Statement)
mkCache = id
#end
```

Depending on how many `extra-deps` this cascades to for our end-users, it could be worth it. I'm not sure, but if you want to go for it I can help more concretely tomorrow.

Reply:
Works for me